### PR TITLE
fix: log dumpers

### PR DESCRIPTION
**Pull request by jamesclark-Zapata**
# The problem

Dumping logs in the CLI no longer worked after we changed logging behaviour a while ago.

Note: this was spotted by Pyright! But it was disabled.

# This PR's solution

Updates the log dumper to work with out + err logs.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: tests/cli/workflow/test_logs.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def test_download_dir_passed(
 
             # Custom mocks
             dumped_path = "<dumped path sentinel>"
-            action._dumper.dump.return_value = dumped_path
+            action._dumper.dump.return_value = (dumped_path,)
```
Comment (alexjuda) [inline tests/cli/workflow/test_logs.py:178]: **nitpick**: isn't this method expected to always return two paths?

**Review by alexjuda — APPROVED** [review]
One minor nitpick

File: tests/cli/workflow/test_logs.py (hunk 1 above)
Comment (jamesclark-Zapata) [reply tests/cli/workflow/test_logs.py]: This better 055f8a7?

**Comment by SebastianMorawiec** [comment]
Nitpick: could you take a look at pyproject.toml pyright exclude list and see if you can remove _logs.py files from there with your changes?

**Pull request merged**